### ISSUE TITLE
populate doesn't work in one to many associations in mongodb

### ISSUE THREAD

**Issue by akshitjain** [balderdashy/sails-mongo#284]
I have create a database in which 2 collections are there and have one to many relation.
Now when we try to use the populate query then we get the result which is not grouped i.e populate doesn't gives the required output as it gives in sails-mysql adapter.
Please reply if anybody have a solution to this

**Comment by akshitjain** [comment]
I got the answer.While we add entries in collections in while using sails-mongo adapter the database created puts an attribute 'id' by default and this is of object type while for sails-mysql it is of integer type.So for this we have to create a separate id attribute ourself set as primary key and type integer.This will solve many of the problems...at least it helped me.

**Comment by particlebanana** [comment]
Hey @akshitjain Waterline will automatically map this out for you, no need to create a custom primary key in mongo and handle all that yourself. There were some issues with populating cross adapter/connection which is what I think you are having issues with (one model is using mysql and the other mongo).

If you could post an example of the two models that are failing to populate that would help.

**Comment by akshitjain** [comment]
var User = Waterline.Collection.extend({
  identity: 'user',
  connection: 'myLocalMongo',
  attributes: {
    first_name: {
      type:'string'
    },
    middle_name:{
      type:'string',
      defaultsTo:'kumar'
      },
    last_name: {
      type:'string'
    },
    age:{
      type:'integer',
      defaultsTo:20
    },
    pets:{
      collection:'pet',
      via:'owner',
      dominant:true
    }
  }
});

var Pet = Waterline.Collection.extend({
  identity: 'pet',
  connection: 'myLocalMongo',
  attributes: {
    name: {
      type:'string'
    },
    breed: {
      type:'string'
    },
    age:{
      type:'integer'
    },
    owner:{
      collection:'user',
      via : 'pets'
  }
  }
});

**Comment by mattcodez** [comment]
I have the same issue. Child will populate with parent but parent will not populate children. If I switch to sails-disk it works fine. Example failing sails-mongo issue here https://gist.github.com/mattcodez/e7cc0e41c787a5d8192e

**Comment by sailsbot** [comment]
Thanks for posting, @akshitjain.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mylastore** [comment]
I can't get this to work either using sails ver 0.12.11 and following the same sample above.
